39/covrebo/test_app.py

```diff
@@ -0,0 +1,121 @@
+from app import app, get_items, _record_exists, _get_item, create_item
+import pytest
+import requests
+from flask import json
+
+url = 'http://127.0.0.1:5000'
+
+@pytest.fixture
+def client():
+    client = app.test_client()
+    return client
+
+def test_get_item():
+    assert _get_item(1) == [{'id': 1, 'name': 'laptop', 'value': 1000}]
+    assert _get_item(2) == [{'id': 2, 'name': 'chair', 'value': 300}]
+    assert _get_item(3) == [{'id': 3, 'name': 'book', 'value': 20}]
+    assert _get_item(4) == []
+    assert _get_item('a') == []
+
+def test_record_exists():
+    assert _record_exists('laptop') == [{'id': 1, 'name': 'laptop', 'value': 1000}]
+    assert _record_exists('chair') == [{'id': 2, 'name': 'chair', 'value': 300}]
+    assert _record_exists('book') == [{'id': 3, 'name': 'book', 'value': 20}]
+    assert _record_exists(4) == []
+    assert _record_exists('a') == []
+
+@pytest.fixture
+def client():
+    client = app.test_client()
+    return client
+
+def test_get_items():
+    res = requests.get(url + '/api/v1.0/items')
+    assert res.status_code == 200
+
+    data = res.json()
+    assert data['items'][0]['id'] == 1
+    assert data['items'][1]['id'] == 2
+    assert data['items'][2]['id'] == 3
+
+def test_get_item():
+    # Verify a successful connection
+    res = requests.get(url + '/api/v1.0/items/1')
+    assert res.status_code == 200
+
+    # Verify correct data
+    data = res.json()
+    assert data['items'][0]['id'] == 1
+
+    # Check for a item outside the list
+    res = requests.get(url + '/api/v1.0/items/42')
+    assert res.status_code == 404
+
+
+def test_create_item(client):
```

Comment:
same: break into multiple test functions

Reply:
I have split the tests into smaller tests, can you review and provide feedback?